modules/navier_stokes/src/variables/BernoulliPressureVariable.C

```diff
@@ -63,47 +65,50 @@ BernoulliPressureVariable::elemIsUpwind(const Elem & elem, const FaceInfo & fi)
 
 bool
 BernoulliPressureVariable::isExtrapolatedBoundaryFace(const FaceInfo & fi,
-                                                      const Elem * const elem) const
+                                                      const Elem * const elem,
+                                                      const Moose::TimeArg & time) const
 {
-  if (isDirichletBoundaryFace(fi, elem))
+  if (isDirichletBoundaryFace(fi, elem, time))
     return false;
   if (!isInternalFace(fi))
     // We are neither a Dirichlet nor an internal face
     return true;
 
   // If we got here, then we're definitely on an internal face
 
-  if (std::get<0>(NS::isPorosityJumpFace(*_eps, fi)))
+  if (std::get<0>(NS::isPorosityJumpFace(*_eps, fi, time)))
     // We choose to extrapolate for the element that is downwind
-    return !std::get<0>(elemIsUpwind(*elem, fi));
+    return !std::get<0>(elemIsUpwind(*elem, fi, time));
 
   return false;
 }
 
 bool
 BernoulliPressureVariable::isDirichletBoundaryFace(const FaceInfo & fi,
-                                                   const Elem * const elem) const
+                                                   const Elem * const elem,
+                                                   const Moose::TimeArg & time) const
```

Comment:
Ahhh... is this the reason why we need the time argument for the `isXYBoundaryFace`? I suppose it is fine in case we plan to support time-dependent boundaries in the future. But if that is not on the table, it feels weird modifying everything for the sake of a function in a very-very specific derived class. Maybe I missed a detail or two so disregard this if it is not the sole reason behind the change.

Reply:
It seems like even now in the Pronghorn-SAM coupling work that @snschune is doing there might be times where you may want to change the boundary that the velocity or pressure Dirichlet condition is being applied to (e.g. what is the inlet and what is the outlet).

Reply:
Ahhh yes, cool stuff!